### PR TITLE
Support current Prescient version

### LINKED ISSUE #1743: [TASK] Support newer versions of prescient
--------
Issue Description
--------

**Is your feature request related to a problem? Please describe.**
New versions of Prescient now include minute by minute data.  This should be handled better by the Prescient Code Interface.

**Describe the solution you'd like**
At least don't crash and output nothing.

**Describe alternatives you've considered**
Keep using older versions of Prescient.

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or task?
- [x] 2. Is it tagged with a priority: critical, normal or minor?
- [x] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [x] 1. If the issue is a defect, is the defect fixed?
- [x] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [x] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [x] 4. If the issue is a defect, does it impact the latest release branch? If yes, is there any issue tagged with release (create if needed)?
- [x] 5. If the issue is being closed without a pull request, has an explanation of why it is being closed been provided?

### PR DESCRIPTION
**Pull request by joshua-cogliati-inl**
--------
Pull Request Description
--------
##### What issue does this change request address? 
Closes #1743

##### What are the significant changes in functionality due to this change request?
Adds ability to still grab Prescient data from new versions.

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?
- [x] 9. If any test used as a basis for documentation examples (currently found in `raven/tests/framework/user_guide` and `raven/docs/workshop`) have been changed, the associated documentation must be reviewed and assured the text matches the example.

### CODE REVIEW

**Comment by joshua-cogliati-inl** [comment]
This was tested with Prescient commit 05492ba7d26207b45b6a14d154defda8bced1e66

File: framework/CodeInterfaces/Prescient/PrescientCodeInterface.py [hunk 1]
```diff
@@ -145,22 +145,32 @@ def _readBusData(self, filename):
...
           first = False
-          if not line.startswith("Date,Hour,Bus,"): # != "Date,Hour,Bus,Shortfall,Overgeneration,LMP,LMP DA":
+          if line.startswith("Date,Hour,Minute,Bus,"):
+            hasMinute = True
+          elif line.startswith("Date,Hour,Bus,"):
+            #line sorta looks like: "Date,Hour,Bus,Shortfall,Overgeneration,LMP,LMP DA"
+            hasMinute = False
+          else:
             assert False, "Unexpected first line of bus detail:" + line
             a = 1/0 #because debug might be disabled
```
Comment (mandd) [inline framework/CodeInterfaces/Prescient/PrescientCodeInterface.py:155]: Is this line correct?

File: framework/CodeInterfaces/Prescient/PrescientCodeInterface.py [hunk 2]
```diff
@@ -179,7 +189,7 @@ def finalizeCodeOutput(self, command, codeLogFile, subDirectory):
       directory = subDirectory
     readFile = os.path.join(directory, toRead)
     if toRead.lower().startswith("hourly"):
-      busData, busList, busDataList = self._readBusData(os.path.join(directory, "bus_detail.csv"))
+      busData, busList, busDataList, hasMinute = self._readBusData(os.path.join(directory, "bus_detail.csv"))
```
Comment (mandd) [inline framework/CodeInterfaces/Prescient/PrescientCodeInterface.py:192]: do we need to check here if hasMinute is available?

File: framework/CodeInterfaces/Prescient/PrescientCodeInterface.py (hunk 1 above)
Comment (joshua-cogliati-inl) [reply framework/CodeInterfaces/Prescient/PrescientCodeInterface.py:155]: Hm, it is at least sorta correct, but there might be a better way to signal an error from a Code Interface. I should look.

File: framework/CodeInterfaces/Prescient/PrescientCodeInterface.py (hunk 2 above)
Comment (joshua-cogliati-inl) [reply framework/CodeInterfaces/Prescient/PrescientCodeInterface.py:192]: No, hasMinute is the check.

File: framework/CodeInterfaces/Prescient/PrescientCodeInterface.py (hunk 1 above)
Comment (joshua-cogliati-inl) [reply framework/CodeInterfaces/Prescient/PrescientCodeInterface.py:155]: How about I change this to something like:
```
raise IOError("Unexpected first line of bus detail in Prescient Code Interface:" + line)
```

Comment (joshua-cogliati-inl) [reply framework/CodeInterfaces/Prescient/PrescientCodeInterface.py]: Changed.

**Comment by mandd** [comment]
PR is good to me. Test dashboard is green. Merging.

**Review by mandd — APPROVED** [review]

**Pull request merged**